src/inc/til/rle.h

```diff
@@ -0,0 +1,1064 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#ifdef UNIT_TESTING
+class RunLengthEncodingTests;
+#endif
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    namespace details
+    {
+        template<typename T, typename S, typename ParentIt>
+        class rle_iterator
+        {
+        public:
+            using iterator_category = std::random_access_iterator_tag;
+            using value_type = T;
+            using pointer = T*;
+            using reference = T&;
+            using size_type = S;
+            using difference_type = typename ParentIt::difference_type;
+
+            // TODO: Enable checked iterators for _ITERATOR_DEBUG_LEVEL != 0.

```

Comment:
Typically we don't allow check-in with just "TODO". We request that it says "TODO GH#<number>" so it's leading to a concrete backlog item. (a.k.a. please file a follow-on item and leave the number here and then grep for the number and remove the todo when you fix it eventually)